### ISSUE TITLE
Login failure

### ISSUE THREAD

**Issue by blaukon** [thobach/MMM-MicrosoftToDo#86]
Hello,

your module worked great, but for about one week I'm getting only errors and couldn't figure it out yet.
I was using your old version and updated to the newest one (after the login didn't work anymore):
```
[2022-08-30 17:03:06.888] [ERROR] Body {
  url: 'https://login.microsoftonline.com/common/oauth2/v2.0/token',
  status: 404,
  statusText: 'Not Found',
  headers:
   Headers {
     _headers:
      { 'set-cookie': [Array],
        date: [Array],
        connection: [Array],
        'content-length': [Array] } },
  ok: false,
  body:
   PassThrough {
     _readableState:
      ReadableState {
        objectMode: false,
        highWaterMark: 16384,
        buffer: BufferList { length: 0 },
        length: 0,
        pipes: null,
        pipesCount: 0,
        flowing: null,
        ended: true,
        endEmitted: false,
        reading: false,
        sync: false,
        needReadable: false,
        emittedReadable: true,
        readableListening: false,
        resumeScheduled: false,
        emitClose: true,
        destroyed: false,
        defaultEncoding: 'utf8',
        awaitDrain: 0,
        readingMore: false,
        decoder: null,
        encoding: null },
     readable: true,
     _events: { prefinish: [Function: prefinish] },
     _eventsCount: 1,
     _maxListeners: undefined,
     _writableState:
      WritableState {
        objectMode: false,
        highWaterMark: 16384,
        finalCalled: false,
        needDrain: false,
        ending: true,
        ended: true,
        finished: true,
        destroyed: false,
        decodeStrings: true,
        defaultEncoding: 'utf8',
        length: 0,
        writing: false,
        corked: 0,
        sync: true,
        bufferProcessing: false,
        onwrite: [Function: bound onwrite],
        writecb: null,
        writelen: 0,
        bufferedRequest: null,
        lastBufferedRequest: null,
        pendingcb: 0,
        prefinished: true,
        errorEmitted: false,
        emitClose: true,
        bufferedRequestCount: 0,
        corkedRequestsFree: [Object] },
     writable: false,
     allowHalfOpen: true,
     _transformState:
      { afterTransform: [Function: bound afterTransform],
        needTransform: false,
        transforming: false,
        writecb: null,
        writechunk: null,
        writeencoding: null } },
  bodyUsed: false,
  size: 0,
  timeout: 0,
  _raw: [],
  _abort: false }
[2022-08-30 17:03:06.920] [ERROR] [MMM-MicrosoftToDo]: getTodos / get access token: https://login.microsoftonline.com/common/oauth2/v2.0/token
```
I've changed my "refresh_token" with a new one, but still the same error.

**Comment by thobach** [comment]
Hi @blaukon, Others had also issues partly after updating to a newer version from a version before 2022, and partly because Microsoft change the maximum duration for some of their client (authentication) keys.
From the logs you shared the issue seems to be with generating the access token. I suggest to try again the whole installation in the Azure portal and update the oauth* configuration parameters accordingly (not just use a new refresh_token). If you like you can also contact me on twitter with a direct message (https://twitter.com/thobach) and I'll let you know my email address to send me screenshots of the Azure portal configuration.

**Comment by blaukon** [comment]
@thobach thank you for your response! I just managed to activate my azure account again, microsoft might have changed the process since the last time. (My account had only been authorized to look at my own project, but not login, I had to create an "app" azure account and add my live account as user).
The newest version of Microsoft Todo is now working fine on a new build, but not yet on my old one - will work on that.
Thank you again!
